packages/next/src/server/next-server.ts

```diff
@@ -1616,16 +1616,20 @@ export default class NextNodeServer extends BaseServer<
       return { finished: true }
     }
 
-    for (let [key, value] of result.response.headers) {
-      if (key.toLowerCase() !== 'set-cookie') continue
+    // Split compound (comma-separated) set-cookie headers
```

Comment:
not a 100% sure what the goal of this section is -- this is my best guess. this change seems to fix the failing test, not sure if it can have any downstream consequences.

Reply:
cc @wyattjoh mind taking a look? you [added this bit](https://github.com/Kikobeats/next.js/commit/e1a397d750df0d1bffea86b867fc2083ef3f3111#diff-c49c4767e6ed8627e6e1b8f96b141ee13246153f5e9142e1da03450c8e81e96fR1891), maybe you have some context for why it was done that way?

Reply:
This shouldn't be required, as the `set-cookie` header when iterated should emit one cookie each. Can we validate this behaviour using a unit test?